### PR TITLE
result: link tests to requirements

### PR DESCRIPTION
**Pull request by Nikhil2206**
## Summary

Link all 256 C++ tests in the `score/result` component to the component requirements using the required `RecordProperty` metadata.

This completes Route 1 traceability authoring for a component that had no pre-existing test requirement metadata.

## Files touched

- `score/result/error_test.cpp`
- `score/result/error_msg_mapping_test.cpp`
- `score/result/result_test.cpp`
- `score/result/details/expected/expected_assignment_test.cpp`
- `score/result/details/expected/expected_constructor_test.cpp`
- `score/result/details/expected/expected_equality_test.cpp`
- `score/result/details/expected/expected_monadic_operator_test.cpp`
- `score/result/details/expected/expected_observer_test.cpp`
- `score/result/details/expected/expected_type_trait_test.cpp`
- `score/result/details/expected/extensions_test.cpp`
- `score/result/details/expected/test_type_traits_test.cpp`
- `score/result/details/expected/unexpected_test.cpp`

## Traceability

Each test now has `PartiallyVerifies`, `Description`, `TestType`, and `DerivationTechnique` metadata.

The links cover `error_handling`, `set_result`, `domain_error_information`, `type_safety`, `std_integration`, `deterministic_behavior`, and `exception_free_operation`.

Observer misuse tests use `fault-injection` and `boundary-values`; systematic ref-qualified and type-variant test families use `equivalence-classes`.

## Validation

- `bazel test --config=bl-x86_64-linux //score/result/...` -> all 7 test targets passed.
- `bazel run //:format.fix -- score/result/details/expected/expected_monadic_operator_test.cpp score/result/details/expected/expected_observer_test.cpp` completed successfully.
- `git diff --check` passed.
- Confirmed 256 tests and 256 `FullyVerifies`/`PartiallyVerifies` links.
- Confirmed all 7 referenced `comp_req` IDs exist in `score/result/docs/requirements/index.rst`.

The full docs build remains blocked by the pre-existing root-owned `_build/` permission error, so requirement ID validation used the documented grep fallback.

## Deliberately left unlinked

None. All 256 result tests have traceability metadata.

## Flagged requirement-quality issue

`extensions_test.cpp` includes `score::cpp::expected` and `score::cpp::optional` interoperability tests in addition to `std::optional` interoperability. These tests are linked to `comp_req__result__std_integration`, whose current wording is narrower and explicitly names `std::optional`. The requirement wording may need clarification in a follow-up change.

### CODE REVIEW

File: score/result/docs/requirements/index.rst [hunk 1]
```diff
@@ -92,6 +92,19 @@ Functional Requirements
...
+   :reqtype: Functional
+   :security: YES
+   :safety: ASIL_B
+   :derived_from: feat_req__baselibs__result_library[version==2]
+   :status: valid
+   :version: 1
+   :tags: inspected
+   :satisfied_by: comp__baselibs_result[version==1]
+
+   The Result library shall provide conversion utilities to transform an expected object into a score::cpp::expected object and vice versa, and to transform an expected's value into a score::cpp::optional<T>. The conversion to score::cpp::optional<T> shall require explicit handling of error cases before discarding error information, ensuring that errors cannot be silently ignored.
```
[the remark this answers is not in the document]
Comment (4og) [reply score/result/docs/requirements/index.rst]: The "expected object" in the requirement text is indeed confusing as it's an implementation detail of Result. I would not mention it in the requirement. It doesn't, however, change the essence of the requirement.

File: score/result/details/expected/expected_observer_test.cpp [hunk 2]
```diff
@@ -231,6 +327,12 @@ TEST(ExpectedTest, CanRetrieveValueFromLValueReference)
 
 TEST(ExpectedTest, AbortsWhenRetrieveValueFromLValueReferenceWithoutValue)
 {
+    RecordProperty("PartiallyVerifies", "comp_req__result__error_handling");
+    RecordProperty("Description",
+                   "Check that value() on an lvalue expected holding an error throws instead of returning a "
+                   "dangling value.");
```
[the remark this answers is not in the document]
Comment (Nikhil2206) [reply score/result/details/expected/expected_observer_test.cpp]: this was a real bug. Verified via expected.h that all four value()/error() ref-qualified overloads gate their throw/abort behavior on the variant's active index rather than on which overload is called, so the tests still passed while silently exercising the rvalue overload instead of the lvalue one they're named for. Removed the stray std::move() at all 4 locations so each test now exercises the overload its name/Description claim to cover.

File: score/result/details/expected/expected_assignment_test.cpp [hunk 3]
```diff
@@ -89,13 +111,25 @@ TEST(ExpectedTest, IsMoveAssignableWithError)
 
 TEST(ExpectedTest, MoveAssignmentHasCorrectNoexcept)
 {
+    RecordProperty("PartiallyVerifies", "comp_req__result__exception_free_operation");
+    RecordProperty("Description",
+                   "Check that expected's move assignment is noexcept only when both the value and error types "
+                   "are themselves nothrow move-assignable.");
```
[the remark this answers is not in the document]
Comment (Nikhil2206) [reply score/result/details/expected/expected_assignment_test.cpp]: Expected.h's operator=(expected&&) noexcept() expression requires both is_nothrow_move_assignable and is_nothrow_move_constructible for T and E, so the Description omitting move-constructibility was incomplete. Updated the wording for both MoveAssignmentHasCorrectNoexcept tests (expected<T,E> and expected<void,E>).

I left the sibling SwapHasCorrectNoexceptSpecification descriptions unchanged though: std::is_nothrow_swappable already subsumes nothrow move-construction (the fallback std::swap's own noexcept-specifier requires both is_nothrow_move_constructible and is_nothrow_move_assignable), so "nothrow swappable" alone is already a complete description there. Pushed

File: score/result/docs/requirements/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nikhil2206) [reply score/result/docs/requirements/index.rst]: Rather than naming score::details::expected (an implementation detail, per @4og's comment above) or spelling out score::Result<T>'s error type, I reworded the requirement to talk about "Result objects"

File: score/result/docs/requirements/index.rst [hunk 4]
```diff
@@ -92,6 +92,19 @@ Functional Requirements
...
 
+.. comp_req:: Future C++ Library Integration
+   :id: comp_req__result__futurecpp_integration
+   :reqtype: Functional
+   :security: YES
+   :safety: ASIL_B
+   :derived_from: feat_req__baselibs__result_library[version==2]
+   :status: valid
+   :version: 1
+   :tags: inspected
```
Comment (4og) [inline score/result/docs/requirements/index.rst]: Please remove the inspected tag. It's supposed to be set after the requirements inspection.

**Review by 4og — CHANGES_REQUESTED** [review]

File: score/result/docs/requirements/index.rst (hunk 4 above)
Comment (Nikhil2206) [reply score/result/docs/requirements/index.rst]: removed

**Review by 4og — APPROVED** [review]
